src/backend/src/lib.rs

```diff
@@ -518,6 +520,29 @@ pub fn add_user_credential(
     }
 }
 
+/// Adds a dApp ID to the user's list of hidden dApp IDs.
+///
+/// # Errors
+/// - Returns `Err` if the ID is already in the `hidden_dapp_ids`, or the user profile is not found, or the user profile version is not up-to-date.
+#[update(guard = "may_write_user_data")]
+pub fn add_user_hidden_dapp_id(
+    request: AddHiddenDappIdRequest,
+) -> Result<(), AddDappSettingsError> {
```

Comment:
Can I provide a 1Mb long app ID?  Sounds as if I might be able to do a DOS fairly cheaply.  See the comment at the end of the PR for a suggested sanity check to call.

Reply:
added a test for the max length of the ID

shall I add something for the max number of IDs that can be saved?

Reply:
Sounds like a good idea. :-) But doesn't have to be in this PR.